lib/terraform/runner.rb

```diff
@@ -9,39 +9,73 @@ class << self
+      # @param input_vars    [Hash]   (optional) key/value pairs as input variables for the terraform-runner run job.
+      # @param tags          [Hash]   (optional) key/value pairs tags for terraform-runner Provisioned resources.
+      # @param credentials   [Array]  (optional) List of Authentication objects for the terraform run job.
+      # @param env_vars      [Hash]   (optional) key/value pairs used as environment variables, for terraform-runner run job.
+      # @param action        [String] (optional) type of action, use ResourceAction::PROVISION or ResourceAction::RETIREMENT.
+      # @param stack_id      [String] (optional) required, if running ResourceAction::RETIREMENT action, used by Terraform-Runner stack_delete job.
+      #
+      # @return [Terraform::Runner::ResponseAsync] Response object of terraform-runner api call
+      def run_async(template_path, input_vars: {}, tags: nil, credentials: [], env_vars: {}, action: ResourceAction::PROVISION, stack_id: nil)
+        case action
+        when ResourceAction::RETIREMENT
+          #  ===== DELETE =====
+          if stack_id.present?
+            _log.debug("Run_aysnc/delete_stack('#{stack_id}') for template: #{template_path}")
+            response = delete_stack_job(
+              stack_id,
+              template_path,
+              :input_vars  => input_vars,
+              :credentials => credentials,
+              :env_vars    => env_vars
+            )
+          else
+            _log.error("'stack_id' is required for #{ResourceAction::RETIREMENT} action")
+            raise "'stack_id' is required for #{ResourceAction::RETIREMENT} action"
+          end
+        else
+          # ===== CREATE =====
+          _log.debug("Run_aysnc/create_stack for template: #{template_path}")
+          response = create_stack_job(
+            template_path,
+            :input_vars  => input_vars,
+            :tags        => tags,
+            :credentials => credentials,
+            :env_vars    => env_vars
+          )
+        end
         Terraform::Runner::ResponseAsync.new(response.stack_id)
       end
 
       # To simplify clients who may just call run, we alias it to call
       # run_async.  If we ever need run_sync, we'll need to revisit this.
       alias run run_async
 
-      # Stop running terraform-runner job by stack_id
+      # To simplify clients who want to create-stack, we alias it to call run_async
+      alias create_stack run_async
+
+      # Delete(destroy) terraform-runner created stack resources.
+      def delete_stack(stack_id, template_path, input_vars: {}, credentials: [], env_vars: {})
+        run_async(template_path,
+                  :input_vars  => input_vars,
+                  :credentials => credentials,
+                  :env_vars    => env_vars,
+                  :action      => ResourceAction::RETIREMENT,
+                  :stack_id    => stack_id)
```

Comment:
What does calling run_async which then calls `delete_stack_job` buy us?  Could you just call `        Terraform::Runner::ResponseAsync.new(delete_stack_job(...))` directly?

Reply:
We originally had run/run_async,  so kept it as is for now..

Reply:
we can do this change in next PR

Reply:
done, pushed the changes